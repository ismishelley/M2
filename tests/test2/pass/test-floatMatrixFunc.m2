int main() {
  matrix float [2][2] m;
  m = [[1.0,1.0];
      [0.0,2.0]];

  printStr("Matrix m is");
  myprint(m);

  printStr("Rows of m");
  printInt(m:rows);
  printStr("");

  printStr("Columns of m");
  printInt(m:cols);
  printStr("");

  printStr("Transpose of m");
  myprint(m:transpose);

  printStr("Trace of m");
  printFloat(m:trace);
  printStr("");

  printStr("Submatrix (0,1,0,1) of m");
  printFloat(m:subMatrix[0,0,0,0]);
  printStr("");

  printStr("Submatrix (0,1,0,1) of m");
  myprint(m:subMatrix[0,1,0,1]);

  return 0;
}

void myprint(matrix float [2][2] M){
  int i;
  int j;
  for(i = 0; i < M:rows; ++i){
    for(j = 0; j < M:cols; ++j){
      printFloat(M[i][j]);
    }
    printStr("");
  }
}

