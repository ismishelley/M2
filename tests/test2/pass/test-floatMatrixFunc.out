Matrix m is
1.000000	1.000000	
0.000000	2.000000	
Rows of m
2	
Columns of m
2	
Transpose of m
1.000000	0.000000	
1.000000	2.000000	
Trace of m
3.000000	
Submatrix (0,1,0,1) of m
1.000000	
Submatrix (0,1,0,1) of m
1.000000	1.000000	
0.000000	2.000000	
